demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -134,220 +550,297 @@ CK_RV xProvisionCertificate( CK_SESSION_HANDLE xSession,
 {
     PKCS11_CertificateTemplate_t xCertificateTemplate;
     CK_OBJECT_CLASS xCertificateClass = CKO_CERTIFICATE;
+    CK_CERTIFICATE_TYPE xCertificateType = CKC_X_509;
     CK_FUNCTION_LIST_PTR pxFunctionList;
     CK_RV xResult;
     uint8_t * pucDerObject = NULL;
     int32_t lConversionReturn = 0;
     size_t xDerLen = 0;
+    CK_BBOOL xTokenStorage = CK_TRUE;
 
+    CK_BYTE xSubject[] = "TestSubject";
```

Comment:
A dummy value?  Do we really need to initialize the entire template in order to process pucCertificate?

Reply:
Should be provided by the caller, then, in my opinion.

Reply:
Conclusion of in person discussion: post-merge, will update xProvisionCertificate to parse the certificate subject out of the cert and add it to the template properly.